src/measurements.jl

```diff
@@ -895,3 +895,77 @@ function _fiducial_WH(::Type{T}, d::Integer) where {T}
 end
 
 _fiducial_WH(d::Integer) = _fiducial_WH(ComplexF64, d)
+
+"""
+    state_discrimination_min_error(
+        ρ::Vector{<:AbstractMatrix{T}},
+        q::Vector{<:AbstractFloat} = fill(1/length(ρ), length(ρ));
+        verbose = false,
+        dualize = false,
+        solver = Hypatia.Optimizer{_solver_type(T)}
+    ) where {T, S<:AbstractFloat}
+
+Return a POVM ``{E_i}_{=1}^N`` and probability P such that if we observe ``E_i`` with average probability P we are in state ``ρ_i``
+
+# Arguments
+- ρ: list of state to be discriminated
+- q: (optional) list of probabilities associated with the state ρ, if not provided, uniform probability is assumed
+"""
+function state_discrimination_min_error(
+    ρ::Vector{<:AbstractMatrix{T}},
+    q::Vector{S} = fill(1/length(ρ),length(ρ));
+    verbose = false,
+    dualize = false,
+    solver = Hypatia.Optimizer{_solver_type(T)}) where {T,S<:AbstractFloat}
+
+    model = JuMP.GenericModel{_solver_type(T)}()    
+    N = length(ρ)
+    d = maximum(size(ρ[1]))
+
+    @assert length(q) == length(ρ) 
+    @assert isapprox(sum(q), 1.0; atol=eps(S))
+
+    E = [JuMP.@variable(model, [1:d, 1:d] in JuMP.HermitianPSDCone()) for i in 1:N]
+    JuMP.@constraint(model, E[N] == I - sum(E[1:N-1]))
+
+    JuMP.@objective(
+        model,
+        Max,
+        sum(q[i] * real(dot(ρ[i]',E[i])) for i in 1:N),
+    )
+
+    if dualize
+        JuMP.set_optimizer(model, Dualization.dual_optimizer(solver; coefficient_type = _solver_type(T)))
+    else
+        JuMP.set_optimizer(model, solver)
+    end
+    !verbose && JuMP.set_silent(model)
+    JuMP.optimize!(model)
+
+    JuMP.is_solved_and_feasible(model) || throw(error(JuMP.raw_status(model)))
+    return [JuMP.value.(e) for e in E], JuMP.objective_value(model)
+end
+export state_discrimination_min_error
+
+"""
+    pretty_good_povm(ρ::Vector{<:AbstractMatrix{T}}) where {T}
+
+Return a POVM `{E_i}_{i=1}^N` using the pretty good measurements method to discriminate between the states `ρ`
+
+Reference: (https://www.math.uwaterloo.ca/~wcleung/co781-presentation-PGM-Lavoie.pdf)
+"""
+function pretty_good_povm(ρ::Vector{<:AbstractMatrix{T}}) where {T}
+    n = length(ρ)
+    d = maximum(size(ρ[1]))
+
+    M = sum(ρ)
+    rootinvM = Hermitian(M)^-0.5
```

Comment:
```suggestion
    rootinvM = Matrix(Hermitian(M)^-0.5)
```
otherwise we might end up with a Hermitian-Hermitian matrix multiplication which is fatal for performance.